### PR TITLE
avoid the downcasting error in Atoi

### PR DESCRIPTION
**Pull request by juniuszhou**

### CODE REVIEW

File: x/ethbridge/client/cli/tx.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 )
 
 func parseNetworkDescriptor(networkDescriptorStr string) (oracletypes.NetworkDescriptor, error) {
```
Comment (smartyalgo) [inline x/ethbridge/client/cli/tx.go:23]: Since we already have verifications here, I think we should use this func where-ever we can.

Comment (juniuszhou) [reply x/ethbridge/client/cli/tx.go:23]: Just forget this method, will use it.

File: x/ethbridge/client/cli/query.go [hunk 2]
```diff
@@ -153,20 +153,20 @@ func GetGlobalSequenceBlockNumber() *cobra.Command {
...
 			}
 
-			networkDescriptor, err := strconv.Atoi(args[0])
+			networkDescriptor, err := oracletypes.ParseNetworkDescriptor(args[0])
 			if err != nil {
 				return err
 			}
 
-			globalSequence, err := strconv.Atoi(args[1])
+			globalSequence, err := strconv.ParseInt(args[1], 10, 64)
```
Comment (smartyalgo) [inline x/ethbridge/client/cli/query.go:161]: Should be parseUint

File: x/ethbridge/client/cli/query.go [hunk 3]
```diff
@@ -192,20 +192,20 @@ func GetProphecyCompleted() *cobra.Command {
...
```
Comment (smartyalgo) [inline x/ethbridge/client/cli/query.go:200]: ParseUint, to avoid casing negative numbers

**Review by smartyalgo — APPROVED** [review]
Comments on GlobalSequence, else LGTM

File: cmd/ebrelayer/txs/parser.go [hunk 4]
```diff
@@ -95,17 +95,14 @@ func BurnLockEventToCosmosMsg(attributes []abci.EventAttribute, sugaredLogger *z
...
 				sugaredLogger.Errorw("network id can't parse", "networkDescriptor", val)
-				return types.CosmosMsg{}, errors.New("network id can't parse")
-			}
-			networkDescriptor = uint32(tempNetworkDescriptor)
-
-			// check if the networkDescriptor is valid
-			if !oracletypes.NetworkDescriptor(networkDescriptor).IsValid() {
 				return types.CosmosMsg{}, errors.New("network id is invalid")
 			}
+			networkDescriptor = tmpNetworkDescriptor
```
[the remark this answers is not in the document]
Comment (juniuszhou) [reply cmd/ebrelayer/txs/parser.go:104]: there is compilation error if use assign to it directly.

File: cmd/ebrelayer/txs/prophecy.go [hunk 5]
```diff
@@ -44,17 +44,13 @@ func ProphecyCompletedEventToProphecyInfo(attributes []abci.EventAttribute, suga
 
 		case types.NetworkDescriptor.String():
 			attributeNumber++
-			tempNetworkDescriptor, err := strconv.ParseUint(val, 10, 32)
+
+			tmpNetworkDescriptor, err := oracletypes.ParseNetworkDescriptor(val)
```
[the remark this answers is not in the document]
Comment (juniuszhou) [reply cmd/ebrelayer/txs/prophecy.go:48]: there is compilation error if use assign to it directly.

File: x/ethbridge/client/cli/query.go (hunk 2 above)
Comment (juniuszhou) [reply x/ethbridge/client/cli/query.go:161]: updated.

File: x/ethbridge/client/rest/rest.go [hunk 6]
```diff
@@ -350,16 +350,16 @@ func getQueryGlobalSequenceBlockNumberHandler(cliCtx client.Context, storeName s
 
 		restNetworkDescriptor := vars[restNetworkDescriptor]
 
-		networkDescriptor, err := strconv.Atoi(restNetworkDescriptor)
+		networkDescriptor, err := oracletypes.ParseNetworkDescriptor(restNetworkDescriptor)
 		if err != nil {
 			rest.WriteErrorResponse(w, http.StatusNotFound, err.Error())
 		}
-		globalSequence, err := strconv.Atoi(restSequence)
+		globalSequence, err := strconv.ParseInt(restNetworkDescriptor, 10, 64)
```
[the remark this answers is not in the document]
Comment (juniuszhou) [reply x/ethbridge/client/rest/rest.go:357]: accepted.

**Pull request merged**